proposals/nnnn-safe-shared-contiguous-storage.md

```diff
@@ -177,6 +201,14 @@ extension UnsafeMutableRawBufferPointer: ContiguousStorage<UInt8> {
 }
 ```
 
+**TODO**: What is the `@_unsafeNonescapableResult` annotation? Would `Slice<UnsafeBufferPointer<UInt8>>` need it?
+
+**TODO**: Do we do a `Sequence.withSpanIfAvailable` API?
+
+**TODO**: What all can we deprecate with this proposal?
```

Comment:
I wasn't expecting any deprecations. We hope this proposal will mean less unsafe code in practice, but it can't eliminate all of it. The `Sequence`&`Collection` updates have a better chance of adding deprecations.

Reply:
This is also relevant to `UTF8Span` potentially replacing `String.withUTF8`.

Reply:
`Array.withUnsafeBytes` and `String.withUTF8` are both interesting cases, because they both expose contiguous storage from types that don't guarantee it – and so they need to temporarily allocate sometimes. This requires a coroutine to handle the state, but that feature isn't explicitly exposed; I'm not sure if we have a solution besides waiting (maybe `@_alwaysEmitIntoClient` would help.)